### ISSUE TITLE
Cannot add or edit challenge

### ISSUE THREAD

**Issue by mk4111** [emfoundation/ce100-app#476]
On clicking 'Submit' of the title/description of a new challenge (i.e. adding a new challenge, on the first page), or on submitting an update on the title/description (i.e. editting a challenge, on the first page), the app breaks with error 500.

![add-edit-chal](https://cloud.githubusercontent.com/assets/13470325/21190794/fe2c0664-c21a-11e6-9d52-f43c586a5083.png)
 
We don't reach the add-tags view, but the updates submitted have been stored, so when you go back to view that challenge, you see that it has been updated.

**Comment by mk4111** [comment]
The challenge gets stored into the database, and gets re-routed to /challenge/{id}/tags which is where it fails.

Need to create a new handlers to display the correct view.

**Comment by JMurphyWeb** [comment]
@mk4111 where did you get up to with this issue? Is it that we forgot to implement that handler?

**Comment by mk4111** [comment]
@JMurphyWeb - yep exactly that. I've got the handler now, but need to test. Will push up once that is done.

**Comment by mk4111** [comment]
Extending to 4h because the issue now extends from add-tags-view (which we were missing), to add-tags handlers.

**Comment by mk4111** [comment]
Current issue fixed, but have found that #488 occurs.

**Comment by SimonLab** [comment]
@mk4111 I think we should create another PR for #488 otherwise your current PR is going a bit big, unless you think #488 is a quick fix?

**Comment by mk4111** [comment]
Yeah - sorry, I only realised it was a separate issue once I dug deeper, at which point I had already started. I can make this two separate PRs though!

**Comment by SimonLab** [comment]
resolved with #488
